core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -143,6 +143,7 @@ private TableProperties() {}
   public static final String PARQUET_COMPRESSION = "write.parquet.compression-codec";
   public static final String DELETE_PARQUET_COMPRESSION = "write.delete.parquet.compression-codec";
   public static final String PARQUET_COMPRESSION_DEFAULT = "gzip";
+  public static final String PARQUET_COMPRESSION_DEFAULT_SINCE_1_4_0 = "zstd";
```

Comment:
Let me think. I do like the name but I also wonder if it sends a message the new default applies to all existing tables that did not provide a default value. It is probably not what happens as we only use this value in new tables.

Reply:
cc @nastra @RussellSpitzer @szehon-ho @rdblue

Reply:
I wonder whether we even need another method, to be honest. Maybe, just update the existing one?

Reply:
I can go either way here.